docs/source/examples/custom_saving.rst

```diff
@@ -0,0 +1,86 @@
+Saving custom fields to the output file
+=======================================
+
+Given the flexibility of the pyDeltaRCM model to modifications via hooks and subclassing, it is necessary that the variables saved to the output netCDF file are similarly customizable.
+Fortunately, the use of subclasses and hooks itself enables flexible setting of gridded variables to be saved as figures, as well as customization of the fields saved to the netCDF file as both variables, and as metadata.
+
+To customize the figures and variables that are saved, the hook ``hook_init_output_file`` should be used to subclass the pyDeltaRCM model.
+
+When adding a model attribute to the list of grids to save as figures, be sure to append the actual *attribute name* to the list of figures to save.
+For example, ``self._save_fig_list.append('active_layer')`` will properly indicate that figures of the active layer should be saved, however ``self._save_fig_list.append('activelayer')`` will not work because the name of the model attribute is ``active_layer`` not ``activelayer``.
```

Comment:
so `save_figs_and_grids` calls `imshow()` on whatever attribute is added to this list, right? We should at least say this here, because a 1d array or scalar will cause an error in imshow, I think

Reply:
Adding an "important" note for this and also have some type-shape error checking so useful errors are thrown